### DISCUSSION TITLE
With Saleor 3.x is it not possible to enable Django Admin

### DISCUSSION THREAD

**Discussion by jorgenader** [saleor/saleor#12046] [General] [answered]
Is it possible and has anyone done it?

---

What I tried was to add both `django.contrib.admin` and `django.contrib.auth` but this resulted in duplicate app installed exception, conflict with `saleor.auth`.

I also tried to monkey-patching `django.contrib.auth` app_label and lift the migrations to local module but this broke after live-reload.

**Comment by maarcingebala** [comment, marked as answer]
Saleor does not support Django Admin. While it is technically possible to enable it as suggested in the Django documentation, it is highly unlikely to work due to the many customizations made that may be incompatible. Instead, we recommend using [Saleor Dashboard](https://github.com/saleor/saleor-dashboard/), a management UI that is more powerful than Django Admin and well-suited for managing data in Saleor, including the complex data dependencies between models that might be hard to manage otherwise. If you need a more low-level database access visual tool, I recommend using pgAdmin or similar tool.

**Reply by Jyrno42** [reply]
What if I have other django packages installed (e.g. blog, django-filer for media upload unrelated to products, etc)? Why is Saleor forcing me to implement standard django modeladmin stuff[1] in the dashboard even if it is totally unrelated to saleor?

> [1] that for most of the packages comes out of the box

**Reply by Jyrno42** [reply]
Additionally, you state in the answer that it is technically possible based on Django documentation but I would say it is not. Django documentation does not show anything about using monkeypatching to trick django into loading its builtin models correctly (since you have added a package that collides with the default auth app namespace). This is something that the OP had to figure out on their own and also isn't really a maintainable solution since it involves fiddling with the internals.

**Reply by maarcingebala** [reply]
Yes, by default Django Admin won't work, but what I mean by "technically possible" is that one might be able to run it with some adjustments to Saleor Core, but I don't know what would be the scope of changes. Saleor has always used a custom dashboard and maintaining compatibility with Django Admain was never our goal.

**Reply by patrys** [reply]
Saleor is not a "Django app." The fact it uses Django underneath is an implementation detail. We do not recommend or support modifying Saleor directly or adding additional packages to the project.
